### PR TITLE
Scavenge thread take 2

### PR DESCRIPTION
**Pull request by osheroff**
@vanchi-zendesk here's a different approach at the scavenge thread.  it's more readable I guess, but I had to expose some guts of RunLoopProcess as well as make it somewhat less accurate.

### CODE REVIEW

File: src/main/java/com/zendesk/maxwell/schema/SchemaScavenger.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+		}
+
+		return list;
+	}
+
+	public void deleteSchema(Long id, Long maxRowsPerSecond) throws SQLException {
+		String[] tables = { "columns", "tables", "databases" };
+
+		try ( Connection connection = connectionPool.getConnection() ) {
+			connection.createStatement().execute("delete from `maxwell`.`schemas` where id = " + id);
```
Comment (vanchi-zendesk) [inline src/main/java/com/zendesk/maxwell/schema/SchemaScavenger.java]: i am a bit worried about deleting the id before we get all the rows. Can't this be done after?

File: src/main/java/com/zendesk/maxwell/schema/SchemaScavenger.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+					long nDeleted = connection.createStatement().executeUpdate(
+						"DELETE FROM maxwell." + tName +
+							" WHERE schema_id = " + id +
+							" LIMIT " + maxRowsPerSecond
+					);
+
+					if (isStopRequested())
+						return;
+
+					if (nDeleted == 0)
```
Comment (vanchi-zendesk) [inline src/main/java/com/zendesk/maxwell/schema/SchemaScavenger.java]: you could put it here, and then break.

File: src/main/java/com/zendesk/maxwell/schema/SchemaScavenger.java (hunk 1 above)
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/schema/SchemaScavenger.java]: yeah.  that was definitely just a coding error when I did the rewrite of this.

**Comment by osheroff** [comment]
@vanchi-zendesk updated

**Comment by vanchi-zendesk** [comment]
:+1:

**Pull request merged**